criu/kerndat.c

```diff
@@ -257,6 +258,45 @@ static int kerndat_get_shmemdev(void)
 	return -1;
 }
 
+static int kerndat_get_secretmem_dev(void)
+{
+#if defined(CONFIG_HAS_MEMFD_SECRET)
+	int fd;
+	void *secretmem;
+	dev_t dev;
+
+	fd = memfd_secret(0);
+	if (fd == -1)
+		return -1;
+
+	if (ftruncate(fd, PAGE_SIZE) < 0) {
+		close(fd);
+		return -1;
+	}
+
+	secretmem = mmap(NULL, PAGE_SIZE, PROT_READ | PROT_WRITE, MAP_SHARED, fd, 0);
+	if (secretmem == MAP_FAILED) {
+		close(fd);
+		return -1;
+	}
+
+	if (kerndat_get_dev(&dev, secretmem, PAGE_SIZE)) {
+		munmap(secretmem, PAGE_SIZE);
+		return -1;
```

Comment:
Missing `close(fd)`. Consider using `goto` for error path.

Reply:
done, thanks!